### ISSUE TITLE
Error with extensions

### ISSUE THREAD

**Issue by sichvoge** [mulesoft/api-workbench#43]
[Enter steps to reproduce below:]

1. i have an extension with the following RAML code

```
#%RAML 1.0 Extension
masterRef: mondo.raml

documentation:
  - title: Introduction
    content: !include ../documentation/introduction.md
  - title: Authentication
    content: !include ../documentation/authentication.md
  - title: Expanding objects
    content: !include ../documentation/expanding-objects.md
```

2. i have a master with the following RAML code

```
#%RAML 1.0
title: Mondo API

traits:
  pagination:
    description: |
      Endpoints which enumerate objects support time-based and curser-based pagination.
    displayName: Pagination
    queryParameters:
      limit:
        type: integer
        maximum: 100
        description: Limits the number of results per-page.
      since:
        type: string
        description: An RFC 3339-encoded timestamp or an object ID.
        examples:
          timestamp:
            content: 2009-11-10T23:00:00Z
          objectID:
            content: tx_00008zhJ3kE6c8kmsGUKgn
      before:
        type: string
        description: An RFC 3339 encoded-timestamp
        example: 2009-11-10T23:00:00Z
```

go to the extension
open RAML Editor Tools
set focus on the documentation node
“Other” tab should show the different entries for the documentation
switch to any other tab than “Others” and return
that is where I get this very messy exception

**Note**: All files are included correctly as for the first time you are on the “Other” tab inside the “Outline” view it shows up the content. Only switching to a different tab and going back fails.

**Atom Version**: 1.1.0
**System**: Mac OS X 10.10.3
**Thrown From**: [api-workbench](https://github.com/mulesoft/api-workbench) package, v0.8.6

### Stack Trace

Uncaught RangeError: Maximum call stack size exceeded

```
At /Users/christianmule/.atom/packages/api-workbench/main.js:0

RangeError: Maximum call stack size exceeded
  at t.directChildren (/Users/christianmule/.atom/packages/api-workbench/main.js:null:null)
  at t.name (/Users/christianmule/.atom/packages/api-workbench/main.js:4:23459)
  at /Users/christianmule/.atom/packages/api-workbench/main.js:4:11160
  at Array.filter (native)
  at t.e.id (/Users/christianmule/.atom/packages/api-workbench/main.js:4:11122)
-----
  at t.getExtractedChildren (/Users/christianmule/.atom/packages/api-workbench/main.js:4:23091)
  at t.children (/Users/christianmule/.atom/packages/api-workbench/main.js:5:3604)
-----

This two lines are getting repeated at least 1M times (guess this is where the stack overflow happens - Had to remove the other lines as it was to much for git.

  at t.elements (/Users/christianmule/.atom/packages/api-workbench/main.js:5:5990)
  at t._findNode (/Users/christianmule/.atom/packages/api-workbench/main.js:5:158)
  at /Users/christianmule/.atom/packages/api-workbench/main.js:5:242
  at Array.forEach (native)
  at t._findNode (/Users/christianmule/.atom/packages/api-workbench/main.js:5:169)
  at t.findElementAtOffset (/Users/christianmule/.atom/packages/api-workbench/main.js:4:23731)
  at e.positionUpdated (/Users/christianmule/.atom/packages/api-workbench/main.js:29:15029)
  at /Users/christianmule/.atom/packages/api-workbench/main.js:29:12900
  at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:86:11)
  at Selection.module.exports.Selection.markerDidChange (/Applications/Atom.app/Contents/Resources/app.asar/src/selection.js:925:29)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/selection.js:41:24
  at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:86:11)
  at Marker.module.exports.Marker.notifyObservers (/Applications/Atom.app/Contents/Resources/app.asar/src/marker.js:262:27)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/marker.js:63:26
  at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:86:11)
  at Marker.module.exports.Marker.emitChangeEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/marker.js:401:20)
  at Marker.module.exports.Marker.update (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/marker.js:346:12)
  at Marker.module.exports.Marker.setHeadPosition (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/marker.js:143:19)
  at Marker.module.exports.Marker.setHeadBufferPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/marker.js:173:32)
  at Marker.module.exports.Marker.setHeadScreenPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/marker.js:183:19)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/cursor.js:65:31
  at Cursor.module.exports.Cursor.changePosition (/Applications/Atom.app/Contents/Resources/app.asar/src/cursor.js:674:7)
  at Cursor.module.exports.Cursor.setScreenPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/cursor.js:63:19)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1317:23
  at TextEditor.module.exports.TextEditor.moveCursors (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1520:9)
  at TextEditor.module.exports.TextEditor.setCursorScreenPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1316:19)
  at TextEditorComponent.module.exports.TextEditorComponent.onMouseDown (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:658:25)
  at HTMLDivElement.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:3:61)

```

### Commands

```
     -7:27 core:save (atom-text-editor.editor.is-focused)
  6x -1:50.5.0 core:backspace (atom-text-editor.editor.is-focused)
     -1:42.9.0 core:save (atom-text-editor.editor.is-focused)
  2x -1:07.3.0 editor:newline (atom-text-editor.editor.is-focused)
     -1:06.2.0 autocomplete-plus:activate (atom-text-editor.editor.is-focused)
     -0:57 autocomplete-plus:confirm (atom-text-editor.editor.is-focused.autocomplete-active)
     -0:54.2.0 core:save (atom-text-editor.editor.is-focused)
     -0:49.3.0 core:cut (atom-text-editor.editor.is-focused)
     -0:48.6.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:45 core:paste (atom-text-editor.editor.is-focused)
  2x -0:44.2.0 core:save (atom-text-editor.editor.is-focused)
     -0:36.3.0 tree-view:add-folder (span.name.icon.icon-file-directory)
     -0:31.6.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:24.6.0 tree-view:move (span.name.icon.icon-file-text)
     -0:20.5.0 core:confirm (atom-text-editor.editor.mini.is-focused)
  2x -0:19.3.0 core:save (atom-text-editor.editor.is-focused)
```

### Config

```json
{}
```

### Installed Packages

```coffee
# User
api-workbench, v0.8.6
linter, v1.10.0

# Dev
No dev packages
```

**Comment by ddenisenko** [comment]
@sichvoge Not reproducible as-is. Could you please provide contents of introduction.md , authentication.md and expanding-objects.md (and checking that exception is still reproducible with the current contents of that files).

**Comment by sichvoge** [comment]
I will send you the files offline. Although I don't think it's the content as this works fine in the main RAML

**Comment by petrochenko-pavel-a** [comment]
Should be fixed. Fix will be published in 0.8.7
